### ISSUE TITLE
mesecons should use my basic_materials mod 

### ISSUE THREAD

**Issue by VanessaE** [minetest-mods/mesecons#440]
Specifically, I provide silicon and an IC in that mod, which are used by a couple other mods.  I think it would be a good idea for mesecons to use it, too, as well as whatever else my mod has that could be of some benefit.

That mod is here: https://forum.minetest.net/viewtopic.php?f=11&t=21000

If interested, I'll submit a pull request.

**Comment by DS-Minetest** [comment]
Related: https://github.com/minetest-mods/technic/pull/451

**Comment by numberZero** [comment]
Don’t reinvent `default`, please.

**Comment by VanessaE** [comment]
Please don't bring those kinds of arguments here.  Default is only the way it is for legacy reasons.

Logic demands that if multiple mods all have a need for the same item, then one such item should be used from a single mod, instead of inventing an equivalent one for each of those multiple mods.  All of that re-inventing is confusing and wasteful, and depending on an external mod to get a few items is not the end of the world, especially since a non-trivial portion of Minetest users will have it anyway, by way of it being used by several other mods (yes, because of my efforts).

The silicon in my mod came from Home Decor, which originally got it from Mesecons, and Technic has its own silicon lump as well (at least, until minetest-mods/technic#451 is merged).  In other words, three projects need that material, and I've had requests to add something equivalent to fiber and something equivalent to glue, to `basic_materials`. 

The `mesecons_materials` mod in this project consists entirely of just those three items.   The IC in my mod is also descended from mesecons, though I can't remember what it was originally used for.

In other words, I've been asked to effectively import the rest of `mesecons_materials`, so that the items therein can be used without depending on mesecons.

**Comment by poikilos** [comment]
> Related: [minetest-mods/technic#451](https://github.com/minetest-mods/technic/pull/451)
> 
> Mesecons has nearly no dependencies, it's self-contained. This shouldn't be changed imo.
Just something to consider: mesecons is great, and useful for many things, but not everyone who wants a certain material wants to depends on mesecons.

> Don’t reinvent default, please.
@numberZero  Since having to depend on mods just for materials is an issue for many modders, VenessaE's idea may be the best solution. Adding these materials to default now is unlikely, and adding more materials to minetest_game would've been helpful 4-6 years ago, so I don't see your point. If you mean that basic_materials is a negative because many mods would need it, that is already true of default, and default doesn't solve the problems that basic_materials solves (it has materials that many modders need, so it prevents duplicating them in each mod). If default did solve the same problem, then basic_materials wouldn't be a thing, and I presume you wouldn't have any problem with these materials.

**Comment by poikilos** [comment]
> I've been asked to effectively import the rest of mesecons_materials

@VanessaE By whom? This is a community project, is it not?

**Comment by VanessaE** [comment]
> > I've been asked to effectively import the rest of mesecons_materials
> 
> @VanessaE By whom? This is a community project, is it not?

That was several months ago, and I don't remember by whom... but I do remember that more than one person asked, and taken together, I'd have ended up doing precisely as I said (because then, all three items would be in `basic_materials` as well as `mesecons_materials`).

I didn't do so because, while those items would surely be useful to other mods, there'd be no point, because it doesn't look Mesecons would be adapted to use the `basic_materials` versions.

**Comment by Niwla23** [comment]
Good idea, *1

**Comment by BruceMustache** [comment]
I like that
Maybe can put basic_materials as optional and if has that mod will register the craft using their items otherwise the same crafting we have today

**Comment by fluxionary** [comment]
Why is this issue still open, when basic_materials dependencies were added ages ago. They've recently been removed, and that breaks things.

**Comment by KollinsPlays** [comment]
As a user, not a programmer, I have had this cause a conflict with other mods. There is a basic materials recipe for silicon but the mesecons mod overwrites that and the mesecons version of silicon is not useable in other mods' recipes. If there is a problem with using basic materials as a dependency, can something be changed so if that mod is installed, mesecons doesn't break the basic materials silicon recipe or also add some way to convert between the two? I don't think players care about the eccentricities of how it's programmed, but they do notice when things break and as it stands, it's broke.
